lib/matplotlib/cbook/__init__.py

```diff
@@ -2767,3 +2768,52 @@ def _topmost_artist(
     in reverse order.
     """
     return _cached_max(reversed(artists))
+
+
+def _define_aliases(alias_d, cls=None):
+    """Class decorator for defining property aliases.
+
+    Use as ::
+
+        @cbook._define_aliases({"property": {"alias", ...}, ...})
+        class C: ...
+
+    For each property, if the corresponding ``get_property`` is defined in the
+    class so far, an alias named ``get_alias`` will be defined; the same will
+    be done for setters.  If neither the getter nor the setter exists, an
+    exception will be raised.
+
+    The alias map is stored as the ``_alias_map`` attribute on the class and
+    can be used by `~.normalize_kwargs`.
+    """
+    if cls is None:
+        return functools.partial(_define_aliases, alias_d)
+
+    def make_alias(name):  # Enfore a closure over *name*.
+        def method(self, *args, **kwargs):
+            return getattr(self, name)(*args, **kwargs)
+        return method
+
+    for prop, aliases in alias_d.items():
+        exists = False
+        for prefix in ["get_", "set_"]:
+            if prefix + prop in vars(cls):
+                exists = True
+                for alias in aliases:
+                    method = make_alias(prefix + prop)
+                    method.__name__ = prefix + alias
+                    method.__doc__ = "alias for `{}`".format(prefix + prop)
+                    setattr(cls, prefix + alias, method)
+        if not exists:
+            raise ValueError(
+                "Neither getter nor setter exists for {!r}".format(prop))
+
+    try:
+        # Deepcopy the inherited alias map if any, to avoid modifying the
+        # parent class.
+        cls._alias_map = copy.deepcopy(cls._alias_map)
+    except AttributeError:
+        cls._alias_map = {}
+    for prop, aliases in alias_d.items():
+        cls._alias_map.setdefault(prop, set()).update(aliases)
```

Comment:
Might be good to enforce uniqueness here, to avoid confusion down the line - especially since dictionary iteration order in `_alias_map` is random.

So something like
```
old_aliases = set().union(*cls._alias_map.values())
new_aliases = set().union(*alias_d.values())
to_remove = old_aliases & new_aliases

for prop, aliases in alias_d.items():
    cls._alias_map.setdefault(prop, set()).update(aliases)

for prop, aliases in alias_d.items():
    cls._alias_map[prop] = aliases - to_remove
```

Reply:
Your code is such that cls._alias_map only contains the aliases defined for cls, but not the inherited ones.  So someone who wants to retrieve aliases for a class needs to walk that class' MRO themselves.

My implmentation stores all aliases, inherited or not.  I don't see why it would be a problem.  Changing the meaning of an alias is perfectly well defined as of now (not that we want to encourage it...).

Also note that as of now none of the classes that actually define aliases inherit from one another, and I'd rather wait until a use case shows up before we decide what kind of semantics we want for this.  The whole mechanism is also private (and intended to stay as such) so it can be modified as needed.

Reply:
> Changing the meaning of an alias is perfectly well defined as of now

I don't think that is true. What should the following do?

```python
@add_aliases({'foo': ['bar']})
class Base(object): pass

@add_aliases({'baz': ['bar']})
class Derived(Base): pass

assert Derived._alias_map == {'foo': ['bar'], 'baz': ['bar']}

normalize_kwargs(Derived._alias_map, dict(bar=1))
```

I'd expect either an error, or the result to be `dict(baz=1)`. But your code on early version of python 3 will sometime randomly give `dict(foo=1)`, because `iteritems` can happen in any order.

> I'd rather wait until a use case shows up before we decide what kind of semantics we want for this

Perhaps then you should simply `raise NotImplemented` if the base class defines `_alias_map`?

Reply:
Ah, fair point.  Fixed.